tools/checkswitch/checkswitch.go

```diff
@@ -23,74 +23,77 @@ import (
 	"golang.org/x/tools/go/analysis/singlechecker"
 )
 
-// orderTypes is the preferred order of types in the switch.
-var orderTypes = map[string]int{
+// bsonOrder is the preferred order of case elements in switch statements.
+var bsonOrder = map[string]int{
 	"Document":     1,
 	"documentType": 1,
+	"tagDocument":  1,
 
 	"Array":     2,
 	"arrayType": 2,
+	"tagArray":  2,
 
 	"float64":    3,
 	"doubleType": 3,
+	"tagFloat64": 3,
 
 	"string":     4,
 	"stringType": 4,
+	"tagString":  4,
 
 	"Binary":     5,
 	"binaryType": 5,
+	"tagBinary":  5,
 
-	"ObjectID":     6,
-	"objectIDType": 6,
+	"tagUndefined": 6,
 
-	"bool":     7,
-	"boolType": 7,
+	"ObjectID":     7,
+	"objectIDType": 7,
+	"tagObjectID":  7,
 
-	"Time":         8,
-	"dateTimeType": 8,
+	"bool":     8,
+	"boolType": 8,
+	"tagBool":  8,
 
-	"NullType": 9,
-	"nullType": 9,
+	"Time":         9,
+	"dateTimeType": 9,
+	"tagTime":      9,
 
-	"Regex":     10,
-	"regexType": 10,
+	"NullType": 10,
+	"nullType": 10,
+	"tagNull":  10,
 
-	"int32":     11,
-	"int32Type": 11,
+	"Regex":     11,
+	"regexType": 11,
+	"tagRegex":  11,
 
-	"Timestamp":     12,
-	"timestampType": 12,
+	"tagDBPointer": 12,
 
-	"int64":     13,
-	"int64Type": 13,
+	"tagJavaScript": 13,
 
-	"CString": 14,
-}
+	"tagSymbol": 14,
 
-// orderTags is the preferred order of Tags in the switch.
-// //nolint: mnd // the numbers represent the order.
-var orderTags = map[string]int{
-	"tagFloat64":         1,
-	"tagString":          2,
```

Comment:
See how tagFloat64 and tagString comes after tagArray to match the order with `types` ordering